src/main/java/com/googlecode/download/maven/plugin/internal/WGetMojo.java

```diff
@@ -384,25 +384,34 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                         getLog().warn("Ignoring download failure.");
                     }
                 }
+
                 boolean done = false;
-                while (!done && this.retries > 0) {
+                for (int retriesLeft = this.retries; !done && retriesLeft > 0; --retriesLeft) {
                     try {
                         this.doGet(outputFile);
                         checksums.validate(outputFile);
                         done = true;
+                    } catch (DownloadFailureException ex) {
+
+                        if (ex.getHttpCode() >= 500) {
+                            getLog().warn(ex.getMessage());
+                        } else {
+                            if (this.failOnError) {
+                                throw new MojoExecutionException(ex.getMessage(), ex);
+                            } else {
+                                getLog().warn(ex.getMessage());
+                            }
+                        }
                     } catch (IOException ex) {
                         getLog().warn("Could not get content", ex);
-                        this.retries--;
```

Comment:
@ajarmoniuk don't we need to fail on IOException if `this.failOnError == true` ?

Reply:
True. Applied the same treatment that we do for `DownloadFailureException`.